python/graphstorm/config/argument.py

```diff
@@ -230,8 +231,18 @@ def __init__(self, cmd_args):
         # If model output is configured, save the runtime train config as a yaml file there,
         # and the graph construction config, if one exists in the input
         if hasattr(self, "_save_model_path") and self._save_model_path:
-            # Ensure model output directory exists
-            os.makedirs(self._save_model_path, exist_ok=True)
+            try:
```

Comment:
Do we need to handle the writable error and save configuration errors separately? And do we need to check the `OSError` and `PermissionError` in such fine-grained level?
How about following Python Exception handle guideline: https://docs.python.org/3/tutorial/errors.html#handling-exceptions? 
Use one `try->execept` to enclose both `os.makdir` and `self._save_graph_construct_config`, and then handle each known exception one by one.

Reply:
Right, what we can do is enclose all permissions/writable checks in one function that checks for both errors, and if that one passes, we can proceed with writing the files if they exist. 

That should simplify the work flow.